server/src/main/java/com/epam/aidial/core/server/controller/DeploymentController.java

```diff
@@ -116,11 +116,13 @@ public static Future<Deployment> selectDeployment(ProxyContext context, String i
 
             Application app = proxy.getApplicationService().getApplication(resource).getValue();
 
-            if (filterCustomProperties) {
-                app = CustomApplicationUtils.filterCustomClientPropertiesWhenNoWriteAccess(context, resource, app);
-            }
-            if (modifyEndpoint) {
-                app = CustomApplicationUtils.modifyEndpointForCustomApplication(context.getConfig(), app);
+            if (app.getCustomAppSchemaId() != null) {
```

Comment:
Same "if" for a app from config?

Reply:
It does seem redundant cause CustomApplicationUtils.filterCustomClientProperties and CustomApplicationUtils.modifyEndpointForCustomApplication have this check.

Reply:
It is like different guarantees applied. When I call CustomApplicationUtils.modifyEndpointForCustomApplication I want the endpoint to be modified and the code 
`    public static String getCustomApplicationEndpoint(Config config, Application application) {
        try {
            String schema = getCustomApplicationSchemaOrThrow(config, application);
            JsonNode schemaNode = ProxyUtil.MAPPER.readTree(schema);
           /// some stuff
        } catch (JsonProcessingException | IllegalArgumentException e) {
            throw new ApplicationTypeSchemaProcessingException("Failed to get custom application endpoint", e);
        }` will throw if it can not be modified because of a lack of schema.